contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractKafkaInputOperator.java

```diff
@@ -834,17 +840,16 @@ private boolean isPartitionRequired(int opid, List<KafkaConsumer.KafkaMeterStats
         return false;
       }
 
+      if (!isPartitionBasedOnLoad) {
```

Comment:
Why would this line be involved. If there is new kafka partition, it would return true at line 829. And new kafka partition would be added to waiting list.